ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java

```diff
@@ -140,86 +145,106 @@ public abstract class IgniteAbstractClient extends DB {
+      debug = IgniteParam.DEBUG.getValue(properties);
+      useEmbeddedIgnite = IgniteParam.USE_EMBEDDED.getValue(properties);
+      disableFsync = IgniteParam.DISABLE_FSYNC.getValue(properties);
+      dbEngine = IgniteParam.DB_ENGINE.getValue(properties);
+      storageProfile = IgniteParam.STORAGE_PROFILES.getValue(properties).toLowerCase();
+
+      // backward compatibility of setting 'dbEngine' as storage engine name only.
+      if (storageProfile.isEmpty() && !dbEngine.isEmpty()) {
+        storageProfile = dbEngine.toLowerCase();
+      }
 
-        if (useEmbeddedIgnite) {
-          initEmbeddedServerNode();
-        } else {
-          initIgniteClientNode();
-        }
-      } catch (Exception e) {
-        throw new DBException(e);
+      replicas = IgniteParam.REPLICAS.getValue(properties);
+      partitions = IgniteParam.PARTITIONS.getValue(properties);
+
+      String workDirProperty = IgniteParam.WORK_DIR.getValue(properties);
+      embeddedIgniteWorkDir = Paths.get(workDirProperty);
+
+      cacheName = properties.getProperty(CoreWorkload.TABLENAME_PROPERTY,
+          CoreWorkload.TABLENAME_PROPERTY_DEFAULT);
+      fieldCount = Integer.parseInt(properties.getProperty(
+          CoreWorkload.FIELD_COUNT_PROPERTY, CoreWorkload.FIELD_COUNT_PROPERTY_DEFAULT));
+      fieldPrefix = properties.getProperty(CoreWorkload.FIELD_NAME_PREFIX,
+          CoreWorkload.FIELD_NAME_PREFIX_DEFAULT);
+      recordsCount = parseLongWithModifiers(properties.getProperty(Client.RECORD_COUNT_PROPERTY,
+          Client.DEFAULT_RECORD_COUNT));
+      batchSize = parseLongWithModifiers(properties.getProperty(Client.BATCH_SIZE_PROPERTY,
+          Client.DEFAULT_BATCH_SIZE));
+
+      for (int i = 0; i < fieldCount; i++) {
+        FIELDS.add(fieldPrefix + i);
       }
-    }
-  }
 
-  private void initIgniteClientNode() throws DBException {
-    igniteClient = IgniteClient.builder().addresses(hosts.split(",")).build();
-    ignite = igniteClient;
-    createTestTable(ignite);
-    kvView = ignite.tables().table(cacheName).keyValueView();
-    rView = ignite.tables().table(cacheName).recordView();
+      hosts = properties.getProperty(HOSTS_PROPERTY);
 
-    if (kvView == null) {
-      throw new DBException("Failed to find cache: " + cacheName);
+      if (ignite == null && !useEmbeddedIgnite && hosts == null) {
+        throw new DBException(String.format(
+            "Required property \"%s\" is missing for Ignite Cluster",
+            HOSTS_PROPERTY));
+      }
+    } catch (Exception e) {
+      throw new DBException(e);
     }
   }
 
-  private void initEmbeddedServerNode() throws DBException {
-    igniteServer = startEmbeddedNode();
-    ignite = igniteServer.api();
+  /**
+   * - Start embedded Ignite node (if needed).
+   * - Get Ignite client (if needed).
+   * - Create test table.
+   *
+   * @param isEmbedded Whether to start embedded node.
+   */
+  private void initIgniteClientNode(boolean isEmbedded) throws DBException {
```

Comment:
Nice to have a better name or extract createTable.

Reply:
updated